### PR TITLE
add set validator net addr

### PR DESCRIPTION
**Pull request by cryptoAtwill**
Add json rpc server and command line for setting the validator net address subnet actor call.

### CODE REVIEW

File: src/cli/commands/manager/net_addr.rs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+#[derive(Debug, Args)]
+#[command(about = "Set the validator net address")]
+pub(crate) struct SetValidatorNetAddrArgs {
+    #[arg(long, short, help = "The JSON RPC server url for ipc agent")]
+    pub ipc_agent_url: Option<String>,
+    #[arg(
+        long,
+        short,
+        help = "The address that needs to set the validator net address"
```
Comment (adlrocha) [inline src/cli/commands/manager/net_addr.rs:57]:
```suggestion
        help = "Owner address of the validator being updated"
```

File: src/cli/commands/manager/net_addr.rs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    )]
+    pub from: Option<String>,
+    #[arg(long, short, help = "The subnet to set the validator")]
+    pub subnet: String,
+    #[arg(long, short, help = "The validator net address")]
```
Comment (adlrocha) [inline src/cli/commands/manager/net_addr.rs:62]:
```suggestion
    #[arg(long, short, help = "New validator net address")]
```

File: src/manager/lotus.rs [hunk 3]
```diff
@@ -219,6 +219,34 @@ impl<T: JsonRpcClient + Send + Sync> SubnetManager for LotusSubnetManager<T> {
         Ok(())
     }
 
+    async fn set_validator_net_addr(
+        &self,
+        subnet: SubnetID,
+        from: Address,
+        validator_net_addr: String,
+    ) -> Result<()> {
+        if !self.is_network_match(&subnet).await? {
```
Comment (adlrocha) [inline src/manager/lotus.rs:228]: This call should go to the parent subnet of the subnetID for which the validator is being changed, I don't know if this check is correct in this case.

File: src/server/handlers/manager/net_addr.rs [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    }
+}
+
+#[async_trait]
+impl JsonRPCRequestHandler for SetValidatorNetAddrHandler {
+    type Request = SetValidatorNetAddrParams;
+    type Response = ();
+
+    async fn handle(&self, request: Self::Request) -> anyhow::Result<Self::Response> {
+        let conn = match self.pool.get(&request.subnet) {
```
Comment (adlrocha) [inline src/server/handlers/manager/net_addr.rs:39]: Here you should get the parent of the subnet for which the validator is being changed.

**Review by adlrocha — APPROVED** [review]
A fre minor bugs around the target subnet where the method needs to be sent. Once you fix those and after testing feel free to merge. You should have the latest version of Lotus merged here: https://github.com/consensus-shipyard/lotus/pull/129

File: src/manager/lotus.rs (hunk 3 above)
Comment (cryptoAtwill) [reply src/manager/lotus.rs:228]: resolved

File: src/server/handlers/manager/net_addr.rs (hunk 4 above)
Comment (cryptoAtwill) [reply src/server/handlers/manager/net_addr.rs:39]: resolved

**Pull request merged**